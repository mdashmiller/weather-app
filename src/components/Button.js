import styled from "styled-components"

const Button = styled.button`
	height: 37px;
    width: 50px;
    margin-left: 8px;
    border-radius: 4px;
    border: 2px solid #c3c3c3;
    color: #c3c3c3;
    background: #4c7d98;
    padding: 0;
    font-size: 18px;
    font-family: inherit;
    font-weight: 900;
    cursor: pointer;

    	&:hover {
    		background: #c3c3c3;
    		color: #4c7d98;
    		border: none;
    		outline: none;
    	} 

    	&:active,
    	:focus,
    	:visited {
    		outline: none;
    	}
`

export default Button
