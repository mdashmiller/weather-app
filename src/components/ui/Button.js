import styled from "styled-components"

const Button = styled.button`
	height: 37px;
    width: 50px;
    margin-left: 8px;
    border-radius: 4px;
    border: 2px solid #eee;
    color: #eee;
    background: #52606b;
    padding: 0;
    font-size: 18px;
    font-family: inherit;
    font-weight: 900;
    cursor: pointer;

    	&:hover {
    		background: #f2b632;
    		border: none;
    		outline: none;
    	} 

    	&:active,
    	:focus,
    	:visited {
    		outline: none;
    	}
`

export default Button
