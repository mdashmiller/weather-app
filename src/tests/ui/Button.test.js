import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Button from '../../components/ui/Button'

describe('<Button> rendering', () => {

	it('renders without crashing', () => {
	  	const div = document.createElement('div')
	  	ReactDOM.render(<Button>Test</Button>, div)
	  	ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<Button>Test</Button>
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

})
