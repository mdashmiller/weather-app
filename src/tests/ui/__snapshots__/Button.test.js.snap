// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> rendering has a valid snapshot 1`] = `
<button
  className="sc-bdVaJa bRkysv"
>
  Test
</button>
`;
